File: SCEUCN-APP/app/src/main/java/cl/ucn/disc/pdis/sceucn/model/Persona.java
/*
 * Created by Christian Farias on 21-04-19 1:30
 * Copyright (c) 2019 . All rights reserved.
 * Last modified 21-04-19 1:29
 */

package cl.ucn.disc.pdis.sceucn.model;

import lombok.Data;
import lombok.NonNull;

/**
 * La persona duenia de un vehiculo.
 */
@Data
public class Persona {

    /**
     * El rut de la persona. Ej: '12.123.132-3'.
     */
    @NonNull
    private String rut;

    /**
     * Los nombres de la persona. Ej: 'Juan Andres'.
     */
    @NonNull
    private String nombres;

    /**
     * Los apellidos de la persona. Ej: 'Visalovic Terreas'.
     */
    @NonNull
    private String apellidos;

    /**
     * El correo electronico de la persona. Ej: 'jisalovic@ucn.cl'.
     */
    private String email;

    /**
     * El numero telefonico. Ej. '+569 1234 5678'.
     */
    private String telefono;

    /**
     * Unidad a la que pertenece la persona.
     * Ej: 'Departamento de Ingenieria de Sistemas y Computacion'.
     */
    private String unidad;

    /**
     * Oficina de esta persona. Ej: 'Pabellon Y1, Oficina 305'.
     */
    private String oficina;

    /**
     * El rol que desempenia esta persona.
     */
    private Rol rol;

    /**
     * Tipo de contrato.
     */
    private Contrato contrato;
}

File: SCEUCN-SERVER/Model/Persona.cs


namespace SCEUCN_SERVER.Model
{
    public enum Contrato { Oficial, Fijo, Temporal }

    public class Persona
    {
        
        public int Id { get; set; }
        
        public string Rut { get; set; }

        public string Nombres { get; set; }

        public string Apellidos { get; set; }

        public string Email{ get; set;}

        public string Movil{ get; set;}

        public string Unidad{ get; set;}

        public string Oficina{ get; set;}

        public string Anexo{ get; set;}

        public Rol Rol{ get; set;}

        public Contrato Contrato{ get; set;}
    }
}